### PR TITLE
dev-python/*: bump semantic_version and dnspython + dependencies

### PR DESCRIPTION
**Pull request by vaartis**
A continuation of #20795. 

@mgorny, I was able to bump django and all dependencies to 3.10 (with some patches), and now the semantic_version tests for django pass for me. Still not sure about the dnspython test fails, since they work fine for me.

### CODE REVIEW

**Comment by mgorny** [comment]
Every day a new challenge, right? ;-)

**Comment by vaartis** [comment]
> Every day a new challenge, right? ;-)

So it seems, so it seems. For constantly, the problem is that the latest twisted version doesn't have an `alpha` keyword, but other versions do. For twisted, there are some dependency problems, I'll try setting all use flags for everything and checking what emerge says. Same thing for tornado.

**Comment by vaartis** [comment]
I've solved the problems that existed for twisted itself. For tornado and constantly, we need an ~alpha keyword on the latest twisted version, could this somehow be requested? The old twisted versions seem to be pretty broken and their tests also probably depended on an older version of pyopenssl which were removed from the tree recently. I could just disable these tests and things would probably work out, but having the ~alpha keyword on the newer version would solve this too, and the old version could be left with python 3.9.

**Comment by vaartis** [comment]
Seems like pyopenssl also requires ~riscv from cryptography, the new version of which does not have that keyword, but the old does,

**Comment by mgorny** [comment]
[Rekeywording of twisted](https://bugs.gentoo.org/773451) is already requested. We just have to wait.

My suggestion would be to generally revbump and drop keywords if really necessary, or just defer adding Python 3.10 for now.

**Comment by mgorny** [comment]
Also, if you managed to split the changes into smaller PRs that would be helpful. Ideally do the packages that don't cause problems first.

**Comment by vaartis** [comment]
> [Rekeywording of twisted](https://bugs.gentoo.org/773451) is already requested. We just have to wait.
> 
> My suggestion would be to generally revbump and drop keywords if really necessary, or just defer adding Python 3.10 for now.

I'll subscribe to the rekeywording thread and wait a bit, then. In the meanwhile, i'll probably move twisted and everything that needs later in the commit history, so everything before it could be merged.

As for cryptography, I was able to get a riscv qemu user-emulator working and am currently testing cryptography there. [The RISC-V project page](https://wiki.gentoo.org/wiki/Project:RISC-V) says that is enough for keywording as of now.

**Comment by vaartis** [comment]
Checked cryptography and pytest-subtests (since cryptography needed it for tests) in qemu-user riscv chroot, both worked fine and tests passed, so I added ~riscv for them.

**Comment by vaartis** [comment]
Separated everything that needs twisted into #20835. The rest don't need it, or at least pkgcheck doesn't find any problems with that.

File: dev-python/pytest-asyncio/pytest-asyncio-0.15.1.ebuild
Comment (mgorny) [inline dev-python/pytest-asyncio/pytest-asyncio-0.15.1.ebuild:28]: I'd rather not do that, unless upstream actually blesses it. Then, it would be more readable as a patch.

**Review by mgorny — CHANGES_REQUESTED** [review]
Also, please sort commits dependency-first.

File: dev-python/pytest-asyncio/pytest-asyncio-0.15.1.ebuild
Comment (mgorny) [reply dev-python/pytest-asyncio/pytest-asyncio-0.15.1.ebuild:28]: Hmm, I see upstream's working hard on a major change, so I suppose this is fine after all.

**Comment by mgorny** [comment]
Ok, I'm completely lost as to which PR is current and what I'm supposed to do with it all ;-). Could you split the packages that are not blocked by anything out, and make a small PR with them specifically?

File: dev-python/pytest-asyncio/pytest-asyncio-0.15.1.ebuild
Comment (vaartis) [reply dev-python/pytest-asyncio/pytest-asyncio-0.15.1.ebuild:28]: Should I still make it a patch instead?

**Comment by vaartis** [comment]
> Ok, I'm completely lost as to which PR is current and what I'm supposed to do with it all ;-). Could you split the packages that are not blocked by anything out, and make a small PR with them specifically?

This is the PR with those not blocked, the blocked ones are in the one mentioned a few comments above. I'll sort the things dependency-first in a bit.
The one being blocked is based on this one, so it has everything this one has + blocked deps, I'll rebase when this is merged, so it'll only have the ones blocked by twisted.

File: dev-python/pytest-asyncio/pytest-asyncio-0.15.1.ebuild
Comment (mgorny) [reply dev-python/pytest-asyncio/pytest-asyncio-0.15.1.ebuild:28]: Yes, please.

**Comment by vaartis** [comment]
Sorted dependency-first to the best of my ability and converted the sed config change to a patch.

**Comment by vaartis** [comment]
Will do the same thing in a moment for jinja.

**Comment by vaartis** [comment]
Alright, that's done. Going to rebase the twisted PR on this one again.

**Comment by mgorny** [comment]
Please skip jinja for now. I've allowed pytest-asyncio because upstream confirmed that it's a problem with their tests and not the package itself. In case of jinja, the buggy calls occur inside the main code.

**Comment by vaartis** [comment]
Moved it to the twisted PR for now, and removed from here.

File: dev-python/urllib3/urllib3-1.26.4.ebuild
Comment (mgorny) [inline dev-python/urllib3/urllib3-1.26.4.ebuild:61]: I'm sorry but this is a big *no* from me. There is clearly some breakage involved here, and I would really like to have upstream at least look into it first.

File: dev-python/selenium/selenium-3.141.0-r1.ebuild
Comment (mgorny) [inline dev-python/selenium/selenium-3.141.0-r1.ebuild:1]: Had to skip it because of urllib3.

File: dev-python/semantic_version/semantic_version-2.8.5.ebuild
Comment (mgorny) [inline dev-python/semantic_version/semantic_version-2.8.5.ebuild:3]: Skipped it because it still fails for me.

File: dev-python/hpack/hpack-3.0.0.ebuild
Comment (mgorny) [inline dev-python/hpack/hpack-3.0.0.ebuild:4]: I've backported the changes from 4.0.0 too to make this a more modern ebuild.

File: dev-python/hpack/hpack-3.0.0.ebuild
Comment (mgorny) [inline dev-python/hpack/hpack-3.0.0.ebuild:39]: We don't do seds on package using pytest anymore, we use `--deselect` instead. Done that.

File: dev-python/hyper-h2/hyper-h2-3.2.0.ebuild
Comment (mgorny) [inline dev-python/hyper-h2/hyper-h2-3.2.0.ebuild:35]: I've also had to add this patch to 4.0.0.

File: dev-python/hyper-h2/hyper-h2-3.2.0.ebuild
Comment (mgorny) [inline dev-python/hyper-h2/hyper-h2-3.2.0.ebuild:40]: Same and same (`--deselect`, updates from newer version).

File: dev-python/hyper-h2/files/hyper-h2-3.2.0-failed-healthcheck.patch
Comment (mgorny) [inline dev-python/hyper-h2/files/hyper-h2-3.2.0-failed-healthcheck.patch:74]: Missing LF at EOF.

File: dev-python/cryptography/cryptography-3.3.2.ebuild
Comment (mgorny) [inline dev-python/cryptography/cryptography-3.3.2.ebuild:3]: Now I'm confused. This ebuild shouldn't be needed by anything. In fact, I think I can remove the old version already.

**Review by mgorny — COMMENTED** [review]
I've merged what I could, please rebase and we'll see where we can go from there.

Note that you're going to see some merge conflicts due to my changes to the ebuilds.

File: dev-python/cryptography/cryptography-3.3.2.ebuild
Comment (vaartis) [reply dev-python/cryptography/cryptography-3.3.2.ebuild:3]: Sorry, did not actually look into reverse-dependencies. If we don't need it anymore, that's good.

File: dev-python/urllib3/urllib3-1.26.4.ebuild
Comment (vaartis) [reply dev-python/urllib3/urllib3-1.26.4.ebuild:61]: Seems like they already know about it: https://github.com/urllib3/urllib3/pull/2209. So we'll just have to wait until this is fixed, I guess..

**Comment by vaartis** [comment]
Moved semantic_version to the twisted PR, because (and i completely forgout about this) it depends on django for tests and that one's in the other PR. For now, just waiting on urllib3 to deal with that test failure, I've subscribed to the PR they have.

File: dev-python/urllib3/urllib3-1.26.4.ebuild
Comment (vaartis) [reply dev-python/urllib3/urllib3-1.26.4.ebuild]: Alright, I applied the patch mentioned there and the tests mentioned are now passing.

File: dev-python/urllib3/urllib3-1.26.4.ebuild
Comment (mgorny) [inline dev-python/urllib3/urllib3-1.26.4.ebuild:65]:
```suggestion
	)
	[[ ${EPYTHON} == python3.10 ]] && deselect+=(
		# Fail because they rely on warnings and there are new deprecation warnings in 3.10
```

**Review by mgorny — CHANGES_REQUESTED** [review]
Thanks for doing this! I've planned on doing it myself but you saved me the work ;-).

File: dev-python/urllib3/urllib3-1.26.4.ebuild
Comment (vaartis) [reply dev-python/urllib3/urllib3-1.26.4.ebuild:67]: Amended

**Comment by mgorny** [comment]
Thanks. I'm going to do a few final touches and merge it!

**Comment by mgorny** [comment]
(this will take some more time as I'm building kernels right now)

**Pull request closed without merging**
